drivers/SmartThings/matter-switch/src/test/test_matter_switch_device_types.lua

```diff
@@ -425,6 +426,8 @@ test.register_coroutine_test(
 test.register_coroutine_test(
     "Test profile change on init for water valve parent cluster as server",
     function()
+        test.socket.device_lifecycle:__queue_receive({ mock_device_water_valve.id, "doConfigure" })
+        mock_device_water_valve:expect_metadata_update({ provisioning_state = "PROVISIONED" })
```

Comment:
It seems like the test was written to have a profile switch occur during the `device_init`/`device_added` event, but that was not happening, and the profile switch only occurs during doConfigure. @hcarter-775 what is the correct behavior?

Reply:
correct behavior is that this profiling should be happening during doConfigure.

Reply:
looks like this was a copy-paste error since the 1.3 update added some new logic to this driver